### ISSUE TITLE
Spring-loaded folders

### ISSUE THREAD

**Issue by eljefuri** [elementary/files#688]
Spring-loaded folders _allow exploration through hierarchies of folders while holding a selection_. See here for an example:

https://macmost.com/using-spring-loaded-folders.html

Note the nuance of what the narrator is pointing out: _"what if I am still not sure if I want to drop this file into this folder, or if I want to examine what is there first [without losing the selection]"_

Apple more or less got this right when designing the latter versions of Finder, and I think Files should more or less take their example and run with it.

Again, imagine that I have spent pain-staking time selecting a number of files that I wish to move to another location. The absence of any UI feedback (#687) upon drop is already disorienting, but worse is that in the event of drag-and-drop, I need to have already opened the folder I want to move the selection to, or move it multiple times (or of course, copy+paste).

Thanks to the Files developers for their kind consideration of this feature.

**Comment by peteruithoven** [comment]
But. Doesn't this already mostly work this way?
I only notice that column view is a bit inconsistent, it doesn't spring-load deeper than one deep?

**Comment by peteruithoven** [comment]
It might be useful to split this up:
- One issue on spring loading on the sidebar. 
- One issue on spring loading one folder deeper in column view. (that's the only other concrete place I found where it doesn't work, did you find others?)

**Comment by eljefuri** [comment]
@peteruithoven No those two things I mentioned were my primary use case and what I tested. I will play with it some more and see if there are any other possibilities or oddities (perhaps across different open Files windows?).

**Comment by peteruithoven** [comment]
I think I've found another one: spring loading on the path parts, in the location bar?

**Comment by eljefuri** [comment]
@peteruithoven Yes indeed, path parts not being springloaded is confirmed for me as well.

**Comment by jeremypw** [comment]
I agree that spring-loaded folder behaviour could be improved but it needs to be broken down into manageable steps.
e.g.
1. Return to the source folder if drag is cancelled.  (Not if drop occured)
2. Spring-loaded behaviour in subfolders in Column View
3. Spring-loaded behaviour in parent folders in Column View

etc, etc

**Comment by eljefuri** [comment]
Perhaps we could also make a separate issue for more immediate visible feedback on the spring load. For example, even MacOS 7-8-9 had this nice "warning" flash effect when you were about to commit to a spring. It lasted maybe 3/4 of a second; the icon would flash and then it would open the folder.

**Comment by jeremypw** [comment]
Yes, please feel free to open an issue,

**Comment by eljefuri** [comment]
I have opened issue #699 to deal with this.

**Comment by jeremypw** [comment]
Limiting this issue to the Column View. A separate issue should be raised for sidebar so that they can be fixed by separate PRs.
